setup.py

```diff
@@ -116,13 +117,59 @@ def get_setup_kwargs(raw=False):
     return kwargs
 
 
+def avoid_common_setup_errors(raw=False):
```

Comment:
I'd suggest to give this a more specific name, e.g. `check_namespace_clashes`. This makes it more obvious what this method does and if we are to add more checks in a similar direction in the future, we'll probably split that into mutliple functions anyways :)